staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -409,7 +416,33 @@ func (c *converterImpl) ToUnstructured(obj interface{}) (map[string]interface{},
 	if err != nil {
 		return nil, err
 	}
-	return *u, nil
+	return u, nil
+}
+
+// DeepCopyJSON deep copies the passed value, assuming it is a valid JSON representation i.e. only contains
+// types produced by json.Unmarshal().
+func DeepCopyJSON(x map[string]interface{}) map[string]interface{} {
+	return deepCopyJSON(x).(map[string]interface{})
+}
+
+func deepCopyJSON(x interface{}) interface{} {
+	switch x := x.(type) {
+	case map[string]interface{}:
+		clone := make(map[string]interface{}, len(x))
+		for k, v := range x {
+			clone[k] = deepCopyJSON(v)
+		}
+		return clone
+	case []interface{}:
+		clone := make([]interface{}, len(x))
+		for i, v := range x {
+			clone[i] = deepCopyJSON(v)
+		}
+		return clone
+	default:
```

Comment:
if this is public, we should add explicit types here probably. Wdyt? And panic if it's a different type (e.g. `map[string]string`, which will give wrong results now).

Reply:
Agree, done. One more case that should be handled is `nil` for `null` in json. https://godoc.org/encoding/json#Unmarshal

Reply:
Hm, also `json.Number`.

Reply:
have never seen json.Number.